### ISSUE TITLE
Changes to com.apple.screensaver on 10.13 have no effect

### ISSUE THREAD

**Issue by mjmerin** [Microsoft/macos-cookbook#30]
The following screensaver plist modification would only work on 10.12 but not 10.13

```
plist 'disable screensaver' do
  path "/Users/#{node['macos']['admin_user']}/Library/Preferences/ByHost/com.apple.screensaver.#{hardware_uuid}.plist"
  entry 'idleTime'
  value 0
end
```
In 10.13, the only key available in screensaver.plist is tokenRemovalAction. 

In 10.12, the only keys available were askForPassword, askForPasswordDelay, and tokenRemovalAction, so setting idleTime would not have worked either. 

Setting these using 

`defaults write com.apple.screensaver askForPassword -bool TRUE`

does a write to the plist file, but does not actually result in ANY observable difference. 

```
abc@defg:~$defaults read com.apple.screensaver
{
    askForPassword = 1;
    tokenRemovalAction = 0;
}
```

**Comment by americanhanko** [comment]
The domain that `defaults` quieries is `~/Library/com.apple.screensaver` whereas the plist that actually contains the `idleTime` settings is `~/Library/Preferences/com.apple.screensaver.<harware UUID>.plist` - if you explicitly look at the ByHost preference as `sudo`, you can see the settings we need to change:

```
sudo defaults read ~/Library/Preferences/ByHost/com.apple.screensaver.610593BF-EF4E-5146-9E91-A57EEA970A0A.plist
{
    PrefsVersion = 100;
    idleTime = 600;
    loginWindowIdleTime = 0;
    moduleDict =     {
        displayName = Flurry;
        moduleName = Flurry;
        path = "/System/Library/Screen Savers/Flurry.saver";
        type = 0;
    };
}
```

The problems is obtaining the Hardware UUID of the machine in the InSpec tests. `system_profiler SPHardwareDataType` is returning the Hardware UUID of the host machine, not the VM.

**Comment by mjmerin** [comment]
More information via the following links:

[Screensaver Security on macOS 10.13 is broken](https://blog.kolide.com/screensaver-security-on-macos-10-13-is-broken-a385726e2ae2)

[Open Radar Bug: askForPassword key unenforceable in 10.13.1 under certain conditions](http://www.openradar.me/35462422)

To quote Victor from the first link:

"If the profile is installed immediately during device bootstrap, the profile enforces the defined setting and the screensaver lock works as expected.

But, if the user has a chance to set their own preference before the profile is applied, the profile has no effect at all. Installing the profile does not override the user’s configuration, but it does lock them out of the ability to change it for themselves.

Adding the askForPasswordDelay key to the profile helps to remediate the profile, but only slightly. The actual preference only appears to be synced after the user logs in again."

".....Michael suggested I log out of the current user, and using another admin account, delete the current user’s Local Items keychain.

After following Michael’s advice and deleting the affected user’s keychain, I logged back in as that user and noted that my broken profile was fixed and enforcing the expected behavior. Great!"
